src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -73,6 +74,10 @@ class DatasetFilter:
     compatible with the REST API. For all parameters capitalization does not
     matter.
 
+    <Tip warning={true}>
+        The `DatasetFilter` class is deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to [`list_datasets`].
+    </Tip>
+
```

Comment:
(otherwise tips are not rendered properly - see https://github.com/huggingface/huggingface_hub/pull/2028#issuecomment-1948712119)

```suggestion
    <Tip warning={true}>

        The `DatasetFilter` class is deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to [`list_datasets`].

    </Tip>

```